awscli/customizations/eks/update_kubeconfig.py

```diff
@@ -119,9 +120,8 @@ def _display_entries(self, entries):
 
     def _run_main(self, parsed_args, parsed_globals):
         client = EKSClient(self._session,
-                           parsed_args.name,
-                           parsed_args.role_arn,
-                           parsed_globals)
+                           parsed_args=vars(parsed_args),
```

Comment:
Let's just accept the [`argparse.Namespace`](https://docs.python.org/3/library/argparse.html#argparse.Namespace) type instead of converting it to a dictionary so that is consistent with the type of `parsed_globals` and other places `parsed_args` is referenced. In the tests, we should be able to just instantiate the `Namespace` class directly. Here is an example of where it is done in a test: https://github.com/aws/aws-cli/blob/4776fb25974c0e1a5f3eeb6ff3f7ff3fe5fc1d97/tests/unit/customizations/cloudtrail/test_validation.py#L897

Reply:
Okay that's cool. I wasn't sure whether `parsed_globals` was a Namespace, it seemed like it wasn't because when I put `vars(parsed_globals)` it gave an error.